src/runtime_src/core/edge/drm/zocl/zocl_ioctl.c

```diff
@@ -66,15 +67,23 @@ zocl_ctx_ioctl(struct drm_device *ddev, void *data, struct drm_file *filp)
 		return zocl_context_ioctl(zdev, data, filp);
 	}
 
-	if (args->op == ZOCL_CTX_OP_OPEN_GCU_FD) {
+	switch(args->op) {
+	case ZOCL_CTX_OP_SK_OPEN:
+		((struct sched_client_ctx *)filp->driver_priv)->sk_cu_id = args->cu_index;
+		break;
+	case ZOCL_CTX_OP_SK_CLOSE:
+		((struct sched_client_ctx *)filp->driver_priv)->sk_cu_id = INVALID_CU_ID;
```

Comment:
Do we need mutex protection here? Or are we relying on softkernel framework to enforce it? If latter, which is not very reliable, we should at least put some comments here